gridcomps/FakeDyn/FakeDynGridComp.F90

```diff
@@ -0,0 +1,163 @@
+#include "MAPL_Generic.h"
+
+module mapl3g_FakeDynGridComp
+
+   use mapl_ErrorHandling
+   use generic3g
+   use mapl3g_FieldCondensedArray, only: assign_fptr_condensed_array
+   use mapl3g_Field_API, only: MAPL_FieldEmptyComplete, VerticalStaggerLoc, VERTICAL_STAGGER_CENTER
+   use mapl3g_UngriddedDims
+   use esmf
+
+   implicit none
+   private
+
+   public :: SetServices
+
+contains
+
+   subroutine SetServices(gridcomp, rc)
+      type(ESMF_GridComp) :: gridcomp
+      integer, intent(out) :: rc
+
+      type(ESMF_HConfig) :: hconfig, collections_config, child_hconfig
+      character(len=:), allocatable :: child_name, collection_name
+      type(ESMF_HConfigIter) :: iter, iter_begin, iter_end
+      logical :: has_active_collections
+      ! class(logger), pointer :: lgr
+      integer :: num_collections, status
+
+      ! Set entry points
+      call MAPL_GridCompSetEntryPoint(gridcomp, ESMF_METHOD_INITIALIZE, init, phase_name="GENERIC::INIT_REALIZE", _RC)
+      call MAPL_GridCompSetEntryPoint(gridcomp, ESMF_METHOD_RUN, run, phase_name="run", _RC)
+
+      _RETURN(_SUCCESS)
+   end subroutine setServices
+
+   subroutine init(gridcomp, importState, exportState, clock, rc)
+      type(ESMF_GridComp) :: gridcomp
+      type(ESMF_State) :: importState
+      type(ESMF_State) :: exportState
+      type(ESMF_Clock) :: clock
+      integer, intent(out)  :: rc
+
+      type(OuterMetaComponent), pointer :: outer_meta
+      type(ESMF_Geom) :: geom
+      class(VerticalGrid), allocatable :: vertical_grid
+      type(ESMF_Field) :: field1, field2
+      integer :: num_levels, status
+
+      outer_meta => get_outer_meta_from_inner_gc(gridcomp,_RC)
+      geom = outer_meta%get_geom()
+      vertical_grid = outer_meta%get_vertical_grid()
+      num_levels = vertical_grid%get_num_levels()
```

Comment:
It might be useful to make this look even more like a true user component.   In particular, I don't want ordinary GEOS users to "see" the outer meta.

Reply:
Is there another way to retrieve `geom`? This is a special case, since this needs to execute before `INIT_USER` phase (where the coupler gets initialized), and at this stage an empty field has been create, but `geom` has not been set yet.

Reply:
For (1) there is a special init phase for the purpose.